### PR TITLE
refactor(app-insights-logger): Don't export logger class

### PR DESCRIPTION
**Pull request by Josmithr**
As a general rule, we prefer not to export classes, as it hinders our ability to make implementation changes making breaking API changes.

This PR replaces the logger class export with a simple factory function.

### CODE REVIEW

**Comment by Josmithr** [comment]
Waiting for the next release to be cut before merging, so this is marked as a draft for now.

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**
